packages/mobile/src/components/CurrencyDisplay.tsx

```diff
@@ -36,44 +49,98 @@ function getBigSymbolStyle(fontSize: number, color: string) {
+    dollarValue = amount.value
+  } else {
+    // Can't convert other currencies for now
+    return null
+  }
+
+  const localValue = convertDollarsToLocalAmount(dollarValue, dollarToLocalRate)
+  if (!localValue) {
+    return null
+  }
+
+  return {
+    value: localValue,
+    currencyCode: localCurrencyCode as string,
+  }
+}
+
+type FormatFunction = (amount: BigNumber.Value, currency?: CURRENCY_ENUM) => string
+
+function getFormatFunction(formatType: FormatType): FormatFunction {
+  switch (formatType) {
+    case FormatType.Default:
+      return getMoneyDisplayValue
+    case FormatType.NetworkFee:
+      return (amount: BigNumber.Value, currency?: CURRENCY_ENUM) =>
+        getNetworkFeeDisplayValue(amount)
+    case FormatType.NetworkFeePrecise:
+      return (amount: BigNumber.Value, currency?: CURRENCY_ENUM) =>
+        getNetworkFeeDisplayValue(amount, true)
+  }
+}
+
 // TODO(Rossy) This is mostly duped by MoneyAmount, converge the two
 export default function CurrencyDisplay({
   type,
   size,
   useColors,
   hideSymbol,
+  showLocalAmount,
   amount,
-  formatAmount,
+  formatType,
   style,
 }: Props) {
   const localCurrencyCode = useLocalCurrencyCode()
-  // tslint:disable-next-line: react-hooks-nesting
-  const localValue = useDollarsToLocalAmount(amount.value) || 0
+  const dollarToLocalRate = useDollarToLocalRate()
+  const goldToDollarRate = useGoldToDollarRate()
 
   const currency =
     amount.currencyCode === CURRENCIES[CURRENCY_ENUM.GOLD].code
       ? CURRENCY_ENUM.GOLD
       : CURRENCY_ENUM.DOLLAR
 
-  // For now only show the local amount when original currency is dollars
-  const localAmount =
-    currency === CURRENCY_ENUM.DOLLAR && localCurrencyCode
-      ? amount.localAmount ?? { value: localValue, currencyCode: localCurrencyCode }
-      : null
-  const displayAmount = localAmount ?? amount
-  const currencySymbol =
-    displayAmount === localAmount
+  // Show local amount only if explicitly set to true when currency is gold
+  const shouldShowLocalAmount = showLocalAmount ?? currency !== CURRENCY_ENUM.GOLD
+  const displayAmount = shouldShowLocalAmount
+    ? getLocalAmount(amount, localCurrencyCode, dollarToLocalRate, goldToDollarRate)
+    : amount
+  const displayCurrency = displayAmount
+    ? displayAmount.currencyCode === CURRENCIES[CURRENCY_ENUM.GOLD].code
+      ? CURRENCY_ENUM.GOLD
+      : CURRENCY_ENUM.DOLLAR
+    : null
+  const currencySymbol = displayAmount
+    ? shouldShowLocalAmount
       ? LocalCurrencySymbol[displayAmount.currencyCode as LocalCurrencyCode]
       : CURRENCIES[currency].symbol
-  const value = new BigNumber(displayAmount.value)
-  const sign = value.isNegative() ? '-' : ''
-  const formattedValue = formatAmount(value.absoluteValue())
+    : null
+  const value = displayAmount ? new BigNumber(displayAmount.value) : null
+  const sign = value?.isNegative() ? '-' : ''
+  const formatAmount = getFormatFunction(formatType)
+  const formattedValue =
```

Comment:
I don't really have a suggestion, but just noting this a a little gnarly with so many ternaries.

Reply:
Yes this will probably evolve and get refactored to a nicer way when we proceed with the merge/refactor of `CURRENCY_ENUM` and `LocalCurrencyCode`.